src/libs/NumberFormatUtils/index.ios.ts

```diff
@@ -0,0 +1,21 @@
+import type {ValueOf} from 'type-fest';
+import intlPolyfill from '@libs/IntlPolyfill';
+import memoize from '@libs/memoize';
+import type CONST from '@src/CONST';
+
+// On iOS, polyfills from `additionalSetup` are applied after memoization, which results in incorrect cache entry of `Intl.NumberFormat` (e.g. lacking `formatToParts` method).
+// To fix this, we need to apply the polyfill manually before memoization.
+// For further information, see: https://github.com/Expensify/App/pull/43868#issuecomment-2217637217
+intlPolyfill();
```

Comment:
What do you think of this structure?:

**src/libs/NumberFormatUtils/initPolyfill.ios.ts**
```
// Add the iOS comment
const initPolyfill = () => {
     intlPolyfill();
}
export default initPolyfill;
```

**src/libs/NumberFormatUtils/initPolyfill.ts**
```
const initPolyfill = () => {}
export default initPolyfill;
```

**src/libs/NumberFormatUtils/index.ts**
```

initPolyfill();

const MemoizedNumberFormat = memoize(Intl.NumberFormat, {maxSize: 10});

function format() {...}

function formatToParts() {...}

export {format, formatToParts};
```

Reply:
Done ✅